.github/workflows/scheduled.yml

```diff
@@ -90,12 +105,26 @@ jobs:
 
     steps:
 
+      - name: Get latest commit from main
+        if: ${{ github.event_name != 'schedule' }}
+        env:
+          GH_TOKEN: ${{ github.token }}
+        id: get-head
+        run: |
+          if [ '${{ github.event_name = 'push' }}' == "true" ]; then
+            # get the parent commit of the current one to get the relevant function signatures
```

Comment:
nit: Not sure i understand whats happening here ; can you explain?

Reply:
So when a `push` event is triggered on a merge `github.sha` is the sha of the merged commit. We of course don't want to compare against that(aka the same commit) but rather the parent commit. So this gets the parent commit sha and uses it to specifically retrieve those functions signatures.

If we are not in a `push` event we just retrieve that sha of the latest commit to main to retrieve the right signatures (or create them if they don't exist, e.g. because the commit was just merged) to match the state of the merge-commit `actions/checkout` will create.